packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,35 @@
 import abi from 'ethereumjs-abi';
 import BN from 'bignumber.js';
 
-function formatValue(value: any): string {
-  if (typeof(value) === 'number' || BN.isBigNumber(value)) return value.toString();
-  else if (typeof(value) === 'string' && value.match(/\d+(\.\d+)?e(\+)?\d+/)) return (new BN(value)).toString(10);
-  else return value;
+export function formatValue(value: any): string | void {
+
+  // Numbers.
+  if(typeof(value) === 'number') {
+    if(value % 1 === 0) return value.toString(10);
```

Comment:
How about when `value` is a very large number? It seems that toString outputs scientific notation:
```
> (2**255).toString(10)
'5.78960446186581e+76'
```

Will the etherumjs abi encoder handle the number in scientific notation?